projects/dvl-fw/src/lib/plugins/default/csv-template-project.ts

```diff
@@ -15,30 +15,49 @@ import { DefaultRecordSet } from './default-record-set';
 
 
 export class CSVTemplateProject extends DefaultProject {
-  private fields: string[];
-  private fieldTypes: {[field: string]: string};
+  private fields: { [suffix: number]: string[] } = {};
+  private fieldTypes: { [suffix: number]: { [field: string]: string } } = {};
 
-  static async create(isiFileContent: string, fileName?: string): Promise<Project> {
-    const project = new CSVTemplateProject(isiFileContent, fileName);
+  static async create(csvFileContents: string[] | string, fileNames?: string[] | string): Promise<Project> {
+    const project = new CSVTemplateProject(csvFileContents, fileNames);
     return project;
   }
 
-  constructor(csvFileContent: string, fileName?: string) {
+  constructor(csvFileContents: string[] | string, fileNames?: string[] | string) {
     super();
-    this.rawData = this.getRawData(csvFileContent);
-    this.dataSources = this.getDataSources();
-    this.recordSets = this.getRecordSets(fileName);
-    this.graphicVariables = this.getGraphicVariables();
+    csvFileContents = isArray(csvFileContents) ? csvFileContents : [csvFileContents];
+    fileNames = isArray(fileNames) ? fileNames : [fileNames];
```

Comment:
Need to also take into account when fileNames is null...
`fileNames = isArray(fileNames) || !fileNames ? fileNames : [fileNames];`